From f45d0608cd7981f84d433566dfa4c2babb4590a7 Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Thu, 1 Nov 2018 17:43:53 +0800
Subject: [PATCH 645/767] ptp: move DPAA2 PTP config to fsl-dpaa2/Kconfig

Move DPAA2 PTP config to fsl-dpaa2/Kconfig.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.12.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/ptp/Kconfig                | 14 --------------
 drivers/staging/fsl-dpaa2/Kconfig  | 13 +++++++++++++
 drivers/staging/fsl-dpaa2/Makefile |  2 +-
 3 files changed, 14 insertions(+), 15 deletions(-)

diff --git a/drivers/ptp/Kconfig b/drivers/ptp/Kconfig
index 0faa6d8c468b..53e31d74502b 100644
--- a/drivers/ptp/Kconfig
+++ b/drivers/ptp/Kconfig
@@ -55,20 +55,6 @@ config PTP_1588_CLOCK_QORIQ
 	  To compile this driver as a module, choose M here: the module
 	  will be called ptp_qoriq.
 
-config PTP_1588_CLOCK_DPAA2
-	tristate "Freescale DPAA2 as PTP clock"
-	depends on FSL_DPAA2_ETH
-	select PTP_1588_CLOCK
-	default y
-	help
-	  This driver adds support for using the DPAA2 1588 timer module
-	  as a PTP clock. This clock is only useful if your PTP programs are
-	  getting hardware time stamps on the PTP Ethernet packets
-	  using the SO_TIMESTAMPING API.
-
-	  To compile this driver as a module, choose M here: the module
-	  will be called dpaa2-rtc.
-
 config PTP_1588_CLOCK_IXP46X
 	tristate "Intel IXP46x as PTP clock"
 	depends on IXP4XX_ETH
diff --git a/drivers/staging/fsl-dpaa2/Kconfig b/drivers/staging/fsl-dpaa2/Kconfig
index e756916c2772..f65d348b2c59 100644
--- a/drivers/staging/fsl-dpaa2/Kconfig
+++ b/drivers/staging/fsl-dpaa2/Kconfig
@@ -52,6 +52,19 @@ config FSL_DPAA2_ETH_DCB
           (PFC) in the driver.
 
           If unsure, say N.
+
+config FSL_DPAA2_PTP_CLOCK
+	tristate "Freescale DPAA2 as PTP clock"
+	select PTP_1588_CLOCK
+	default y
+	help
+	  This driver adds support for using the DPAA2 1588 timer module
+	  as a PTP clock. This clock is only useful if your PTP programs are
+	  getting hardware time stamps on the PTP Ethernet packets
+	  using the SO_TIMESTAMPING API.
+
+	  To compile this driver as a module, choose M here: the module
+	  will be called dpaa2-rtc.
 endif
 
 source "drivers/staging/fsl-dpaa2/mac/Kconfig"
diff --git a/drivers/staging/fsl-dpaa2/Makefile b/drivers/staging/fsl-dpaa2/Makefile
index a921051f0a79..8c3549d72876 100644
--- a/drivers/staging/fsl-dpaa2/Makefile
+++ b/drivers/staging/fsl-dpaa2/Makefile
@@ -5,5 +5,5 @@
 obj-$(CONFIG_FSL_DPAA2_ETH)		+= ethernet/
 obj-$(CONFIG_FSL_DPAA2_MAC)	+= mac/
 obj-$(CONFIG_FSL_DPAA2_EVB)	+= evb/
-obj-$(CONFIG_PTP_1588_CLOCK_DPAA2) += rtc/
+obj-$(CONFIG_FSL_DPAA2_PTP_CLOCK) += rtc/
 obj-$(CONFIG_FSL_DPAA2_ETHSW)	+= ethsw/
-- 
2.17.0

